resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -59,16 +59,18 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   private val kubernetesNamespace = conf.get(KUBERNETES_NAMESPACE)
 
-  private val kubernetesDriverPodName = conf
-    .get(KUBERNETES_DRIVER_POD_NAME)
-    .getOrElse(throw new SparkException("Must specify the driver pod name"))
+  private val kubernetesDriverJobName = conf
+    .get(KUBERNETES_DRIVER_JOB_NAME)
+    .getOrElse(throw new SparkException("Must specify the driver job name"))
   private implicit val requestExecutorContext = ExecutionContext.fromExecutorService(
     requestExecutorsService)
 
-  private val driverPod = kubernetesClient.pods()
-    .inNamespace(kubernetesNamespace)
-    .withName(kubernetesDriverPodName)
-    .get()
+  private val driverPod: Pod = {
+    val pods = kubernetesClient.pods()
+      .inNamespace(kubernetesNamespace).withLabel("job-name", kubernetesDriverJobName).list()
```

Comment:
Actually would like @foxish or @liyinan926 to confirm if that is the case?

Reply:
I don't think this can happen, I can think two scenarios:
- Job fails: No one restarts the Job the user need to use the spark-submit again (all job related pods will be deleted because of the Ownerreference)
- Pod fails: Job will recreate a new Driver Pod to replace the failed one. There will be only one Driver pod because the failed one will be removed by the Kubernetes garbage collector. 

Can you please elaborate on what do you mean by restart here?

Reply:
> Pod fails: Job will recreate a new Driver Pod to replace the failed one. There will be only one Driver pod because the failed one will be removed by the Kubernetes garbage collector.

For this one I'm not sure if the GC is done immediately. If that's part of the Kubernetes contract, then we're fine, if not, then we can't make any assumptions.

Reply:
Okey I will update the filter with an additional check if the found pod is in running state.